pdf/model/fonts/ttfparser.go

```diff
@@ -258,86 +369,345 @@ func (t *ttfParser) ParseCmap() (err error) {
-					return
-				}
-				s = strings.Replace(s, "\x00", "", -1)
-				var re *regexp.Regexp
-				if re, err = regexp.Compile("[(){}<> /%[\\]]"); err != nil {
-					return
-				}
-				t.rec.PostScriptName = re.ReplaceAllString(s, "")
+	if err = t.Seek("name"); err != nil {
+		return
+	}
+	tableOffset, _ := t.f.Seek(0, os.SEEK_CUR)
+	t.rec.PostScriptName = ""
+	t.Skip(2) // format
+	count := t.ReadUShort()
+	stringOffset := t.ReadUShort()
+	for j := uint16(0); j < count && t.rec.PostScriptName == ""; j++ {
+		t.Skip(3 * 2) // platformID, encodingID, languageID
+		nameID := t.ReadUShort()
+		length := t.ReadUShort()
+		offset := t.ReadUShort()
+		if nameID == 6 {
+			// PostScript name
+			t.f.Seek(int64(tableOffset)+int64(stringOffset)+int64(offset), os.SEEK_SET)
+			var s string
+			s, err = t.ReadStr(int(length))
+			if err != nil {
+				return
 			}
+			s = strings.Replace(s, "\x00", "", -1)
+			var re *regexp.Regexp
+			if re, err = regexp.Compile("[(){}<> /%[\\]]"); err != nil {
+				return
+			}
+			t.rec.PostScriptName = re.ReplaceAllString(s, "")
 		}
-		if t.rec.PostScriptName == "" {
-			err = fmt.Errorf("the name PostScript was not found")
-		}
+	}
+	if t.rec.PostScriptName == "" {
+		err = fmt.Errorf("the name PostScript was not found")
 	}
 	return
 }
 
 func (t *ttfParser) ParseOS2() (err error) {
-	err = t.Seek("OS/2")
-	if err == nil {
-		version := t.ReadUShort()
-		t.Skip(3 * 2) // xAvgCharWidth, usWeightClass, usWidthClass
-		fsType := t.ReadUShort()
-		t.rec.Embeddable = (fsType != 2) && (fsType&0x200) == 0
-		t.Skip(11*2 + 10 + 4*4 + 4)
-		fsSelection := t.ReadUShort()
-		t.rec.Bold = (fsSelection & 32) != 0
-		t.Skip(2 * 2) // usFirstCharIndex, usLastCharIndex
-		t.rec.TypoAscender = t.ReadShort()
-		t.rec.TypoDescender = t.ReadShort()
-		if version >= 2 {
-			t.Skip(3*2 + 2*4 + 2)
-			t.rec.CapHeight = t.ReadShort()
-		} else {
-			t.rec.CapHeight = 0
-		}
+	if err = t.Seek("OS/2"); err != nil {
+		return
+	}
+	version := t.ReadUShort()
+	t.Skip(3 * 2) // xAvgCharWidth, usWeightClass, usWidthClass
+	fsType := t.ReadUShort()
+	t.rec.Embeddable = (fsType != 2) && (fsType&0x200) == 0
+	t.Skip(11*2 + 10 + 4*4 + 4)
+	fsSelection := t.ReadUShort()
+	t.rec.Bold = (fsSelection & 32) != 0
+	t.Skip(2 * 2) // usFirstCharIndex, usLastCharIndex
+	t.rec.TypoAscender = t.ReadShort()
+	t.rec.TypoDescender = t.ReadShort()
+	if version >= 2 {
+		t.Skip(3*2 + 2*4 + 2)
+		t.rec.CapHeight = t.ReadShort()
+	} else {
+		t.rec.CapHeight = 0
 	}
 	return
 }
 
+// ParsePost reads the "post" section in a TrueType font table and sets t.rec.GlyphNames.
 func (t *ttfParser) ParsePost() (err error) {
-	err = t.Seek("post")
-	if err == nil {
-		t.Skip(4) // version
-		t.rec.ItalicAngle = t.ReadShort()
-		t.Skip(2) // Skip decimal part
-		t.rec.UnderlinePosition = t.ReadShort()
-		t.rec.UnderlineThickness = t.ReadShort()
-		t.rec.IsFixedPitch = t.ReadULong() != 0
+	if err = t.Seek("post"); err != nil {
+		return
 	}
+	//versionUpper := t.ReadShort()
```

Comment:
These lines should be removed? Are contained in formatType ?

Reply:
Yes. They have been replaced by formatType